lib/orderbook/OrderBook.ts

```diff
@@ -136,6 +151,56 @@ class OrderBook extends EventEmitter {
     }
   }
 
+  public addPair = async (pair: Pair) => {
+    const pairId = derivePairId(pair);
+    if (this.pairs.has(pairId)) {
+      throw errors.PAIR_ALREADY_SUPPORTED(pairId);
+    }
+    if (!this.currencies.has(pair.baseCurrency)) {
+      throw errors.UNSUPPORTED_CURRENCY(pair.baseCurrency);
+    }
+    if (!this.currencies.has(pair.quoteCurrency)) {
+      throw errors.UNSUPPORTED_CURRENCY(pair.quoteCurrency);
+    }
+
+    const pairInstance = await this.repository.addPair(pair);
+    this.pairs.set(pairInstance.id, pairInstance);
+    return pairInstance;
+  }
+
+  public addCurrency = async (currency: CurrencyFactory) => {
+    if (this.currencies.has(currency.id)) {
+      throw errors.CURRENCY_ALREADY_SUPPORTED(currency.id);
+    }
+    const currencyInstance = await this.repository.addCurrency(currency);
+    this.currencies.set(currencyInstance.id, currencyInstance);
+  }
+
+  public removeCurrency = (currencyId: string) => {
+    const currency = this.currencies.get(currencyId);
+    if (currency) {
+      for (const pairId of this.pairIds) {
+        if (pairId.includes(currencyId)) {
+          throw errors.CURRENCY_CANNOT_BE_REMOVED(currencyId, pairId);
+        }
+      }
+      this.currencies.delete(currencyId);
+      return currency.destroy();
```

Comment:
We should declare functions that return a promise as `async`.

Reply:
Stylistically I don't think there's a problem with non-async methods returning promises, I believe there are instances of that throughout the codebase and as far as I know it's not bad practice or anything and doesn't affect the compiler or type checking. Correct me if I'm wrong.

In this case specifically it's actually returning a `Bluebird` and not a promise, they function pretty much the same but aren't exactly the same I believe, and making the method `async` gives it a return type of `Promise<void>` instead of `Bluebird<void>` which seems undesirable to me.

Reply:
We can leave it as is then. I wasn't aware that not declaring `async` doesn't change its behavior.